src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.dao.impl;
+
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Role getByName(String roleNameString) {
+        try (Session session = factory.openSession()) {
+            Role.RoleName roleName = Role.RoleName.valueOf(roleNameString);
+            Query<Role> findByName = session.createQuery(
+                    "SELECT r FROM Role r "
+                            + "WHERE r.name = :name", Role.class);
+            findByName.setParameter("name", roleName);
+            return findByName.uniqueResult();
+        } catch (Exception e) {
+            throw new DataProcessingException("Role with name " + roleNameString + " not found", e);
+        }
```

Comment:
it would be better to make this method Optional<Role> and use uniqueResultOptional instead of uniqueResult. In sevice you can check Optional